crates/services/sync/src/import.rs

```diff
@@ -0,0 +1,363 @@
+//! # Importer Task
+//! This module contains the import task which is responsible for
+//! importing blocks from the network into the local blockchain.
+
+use std::{
+    ops::RangeInclusive,
+    sync::Arc,
+};
+
+use fuel_core_services::{
+    SharedMutex,
+    StateWatcher,
+};
+use fuel_core_types::{
+    blockchain::{
+        block::Block,
+        consensus::Sealed,
+        primitives::BlockHeight,
+        SealedBlock,
+        SealedBlockHeader,
+    },
+    services::p2p::SourcePeer,
+};
+use futures::{
+    stream::{
+        self,
+        StreamExt,
+    },
+    Stream,
+};
+use std::future::Future;
+use tokio::sync::Notify;
+
+use crate::{
+    ports::{
+        BlockImporterPort,
+        ConsensusPort,
+        PeerToPeerPort,
+    },
+    state::State,
+};
+
+#[cfg(test)]
+pub(crate) use tests::empty_header;
+
+#[cfg(test)]
+mod tests;
+
+#[cfg(test)]
+mod back_pressure_tests;
+
+#[derive(Clone, Copy, Debug)]
+/// Parameters for the import task.
+pub struct Config {
+    /// The maximum number of get header requests to make in a single batch.
+    pub max_get_header_requests: usize,
+    /// The maximum number of get transaction requests to make in a single batch.
+    pub max_get_txns_requests: usize,
+}
+
+pub(crate) struct Import<P, E, C> {
+    /// Shared state between import and sync tasks.
+    state: SharedMutex<State>,
+    /// Notify import when sync has new work.
+    notify: Arc<Notify>,
+    /// Configuration parameters.
+    params: Config,
+    /// Network port.
+    p2p: Arc<P>,
+    /// Executor port.
+    executor: Arc<E>,
+    /// Consensus port.
+    consensus: Arc<C>,
+}
+
+impl<P, E, C> Import<P, E, C> {
+    pub(crate) fn new(
+        state: SharedMutex<State>,
+        notify: Arc<Notify>,
+        params: Config,
+        p2p: Arc<P>,
+        executor: Arc<E>,
+        consensus: Arc<C>,
+    ) -> Self {
+        Self {
+            state,
+            notify,
+            params,
+            p2p,
+            executor,
+            consensus,
+        }
+    }
+}
+impl<P, E, C> Import<P, E, C>
+where
+    P: PeerToPeerPort + Send + Sync + 'static,
```

Comment:
[nit] I guess you can make `Send + Sync` default bound for all ports like `PeerToPeerPort: Send + Sync`

Reply:
There's no reason for these traits to be send and sync it's only in this usecase that it's required.